Editor/Window/ManagedEC2Page.cs

```diff
@@ -107,8 +107,37 @@ private void UpdateGUI()
                     or StackStatus.UpdateComplete);
 
             _deploymentScenariosInput.SetEnabled(_deploymentSettings.CanEdit);
-            _fleetParamsInput.SetEnabled(_deploymentSettings.CanEdit); 
-            _ec2DeploymentStatusLabel.text = _deploymentSettings.CurrentStackInfo.StackStatus;
+            _fleetParamsInput.SetEnabled(_deploymentSettings.CanEdit);
+
+            var stackStatus = _deploymentSettings.CurrentStackInfo.StackStatus;
+            var textProvider = new TextProvider();
+            if (stackStatus == null)
+            {
+                _statusIndicator.Set(State.Inactive, textProvider.Get(Strings.ManagedEC2DeployStatusNotDeployed));
+            }
+            else if (stackStatus.IsStackStatusOperationDone())
+            {
+                if (stackStatus.IsStackStatusFailed())
+                {
+                    _statusIndicator.Set(State.Failed, textProvider.Get(Strings.ManagedEC2DeployStatusFailed));
+                }
+                else
+                {
+                    _statusIndicator.Set(State.Success, textProvider.Get(Strings.ManagedEC2DeployStatusDeployed));
+                }
+            }
+            else if (stackStatus == StackStatus.DeleteInProgress)
+            {
+                _statusIndicator.Set(State.InProgress, textProvider.Get(Strings.ManagedEC2DeployStatusDeleting));
+            }
+            else if (stackStatus.IsStackStatusInProgress())
+            {
+                _statusIndicator.Set(State.InProgress, textProvider.Get(Strings.ManagedEC2DeployStatusDeploying));
+            }
+            else
+            {
+                _statusIndicator.Set(State.Inactive, textProvider.Get(Strings.ManagedEC2DeployStatusNotDeployed));
+            }
```

Comment:
Can StackStatus just have methods for COMPLETED and FAILED? That way this whole if block can just be a switch block for different StackStatus types. Makes it more readable and avoids the need for nested if-else conditions